contribs/gmf/src/directives/layertree.js

```diff
@@ -149,87 +149,272 @@ gmf.LayertreeController = function($http, $sce, ngeoCreatePopup,
+
+
+/**
  * Create and return a layer corresponding to the ngeo layertree's node.
- * Currently only creates WMS layers (internal or external) and WMTS layers.
+ * This function will only create a layer for each "top-level" (depth 1) groups.
+ *
+ * On "not mixed" type nodes, the returned layer will be an ol.layer.Image (WMS)
+ * with each name of node's children as LAYERS parameters.
+ *
+ * On "mixed" type node, the returned  layer will be an ol.layer.Group with
+ * a collection of layers that corresponds to each children of the node.
+ *
+ * All layer created will receive:
+ *  - A 'querySourceId' parameter with the node id as value.
+ *  - A 'layerName' parameter with the node name as value.
+ *
+ * All layer created will be added at the top of the map and with a Z Index
+ * value of 1.
+ *
+ * If the node metadata 'isChecked' value is false, the layer visibility will
+ * be set to false.
  * @param {GmfThemesNode} node Layer tree node.
+ * @param {number=} opt_depth ngeo layertree node depth.
+ * @param {boolean=} opt_createWMS True to allow create wms layer.
  * @return {ol.layer.Base} The OpenLayers layer or group for the node.
  * @export
  */
-gmf.LayertreeController.prototype.getLayer = function(node) {
-  var layer;
-  var layerName = node.name;
-  var layerURL = node.url || this.gmfWmsUrl_;
-  var i, children = node.children;
-
-  // If node is a group.
-  if (goog.isDef(children)) {
-    var layers = new ol.Collection();
-    for (i = 0; i < children.length; i++) {
-      layers.push(this.getLayer(children[i]));
+gmf.LayertreeController.prototype.getLayer = function(node, opt_depth,
+        opt_createWMS) {
+  var type = this.getNodeType_(node);
+  var layer = null;
+
+  if (opt_depth === 1) {
+    switch (type) {
+      case gmf.LayertreeController.TYPE_MIXEDGROUP:
+        return this.getLayerCaseMixedGroup_(node);
+      case gmf.LayertreeController.TYPE_NOTMIXEDGROUP:
+        layer = this.getLayerCaseNotMixedGroup_(node);
+        break;
+    }
+    switch (type) {
+      case gmf.LayertreeController.TYPE_WMTS:
+        layer = this.getLayerCaseWMTS_(node);
+        break;
+      case gmf.LayertreeController.TYPE_WMS:
+      case gmf.LayertreeController.TYPE_EXTERNALWMS:
+        var url = node.url || this.gmfWmsUrl_;
+        layer = opt_createWMS ?
+            this.layerHelper_.createBasicWMSLayer(url, node.name) : null;
+        break;
     }
-    return this.layerHelper_.createBasicGroup(layers);
   }
 
-  // If node describes a layer that was already created.
-  layer = this.layerHelper_.findLayer(this.existingLayers_,
-      this.layerHelper_.makeHelperID(layerURL, layerName));
   if (goog.isDefAndNotNull(layer)) {
-    return layer;
+    layer.set('querySourceId', node.id);
+    layer.set('layerName', node.name);
+    // The layer must be upper than the background
+    layer.setZIndex(1);
+    // Add the new layer on the map but behind (before) others layers
+    this.map.getLayerGroup().getLayers().insertAt(0, layer);
```

Comment:
I'm note sur that that's completely the same things... (I'm not sur I've the `insertAt` method with your idea). But I'll try it.